### PR TITLE
Issue #2675836: Variation is not properly factoring attribute depende…

### PR DESCRIPTION
**Pull request by mglaman**
…ncies when selecting from user input

### CODE REVIEW

**Comment by mglaman** [comment]
:) PR, preserves attributes.

fix-attributes-selection

File: modules/product/src/Form/AddToCartForm.php [hunk 1]
```diff
@@ -258,22 +258,26 @@ protected function selectVariationFromUserInput(array $variations, FormStateInte
...
+      // attribute, because it defines our final available variations.
       $attributes = $user_input['attributes'];
-      foreach ($variations as $variation) {
-        $match = TRUE;
-        foreach ($attributes as $field_name => $value) {
+      array_pop($attributes);
+
+      // Loop the through attributes to find qualifying variations.
+      foreach ($attributes as $field_name => $value) {
+        foreach ($variations as $key => &$variation) {
```
Comment (bojanz) [inline modules/product/src/Form/AddToCartForm.php]: You don't actually need the & in front of variation?

Comment (mglaman) [inline modules/product/src/Form/AddToCartForm.php]: Shoot, no. For some reason I has have a derp moment trying to unset the array item :P

**Comment by bojanz** [comment]
Committed in 3201dda9dec2c2b56c8b19bbc9a002c0155b93e9 with comment tweaks.

**Pull request closed without merging**